src/@utils/index.ts

```diff
@@ -17,3 +17,27 @@ export function sortAssets(items: Asset[], sorted: string[]) {
   })
   return items
 }
+
+export const isObjectLike = (object: any) => {
+  return object !== null && typeof object === 'object' && !Array.isArray(object)
+}
+
+export const getObjectPropertyByPath = (object: any, path = '') => {
+  if (!isObjectLike(object)) return object
+  path = path.replace(/\[(\w+)\]/g, '.$1') // convert indexes to properties
+  path = path.replace(/^\./, '') // strip a leading dot
+  const pathArray = path.split('.')
+  for (let i = 0, n = pathArray.length; i < n; ++i) {
+    const key = pathArray[i]
+    try {
+      if (key in object) {
+        object = object[key]
+      } else {
+        return
+      }
+    } catch {
+      return object
```

Comment:
question(non-blocking): It might be better to always return `undefined` when the function fails to get a property instead of in one case `undefined` and in the other the `object`. Any reason for this inconsistency?

Reply:
thought(non-blocking): it's not clear what kind of errors we are expecting here. If we are expecting object as null or undefined, it would be better to check for these conditions explicitly. This makes it easier for the maintainer of the function to understand potential problems here.

Reply:
Agreed on being consistent with return value on failure. Updated accordingly.